src/directives/plugin.ts

```diff
@@ -53,12 +53,28 @@ function runDirectives(directives: {
         try {
           const directive = new directives[token.info](state)
           const data = directiveToData(token, directive)
-          const newTokens = directive.run(data)
-          // Ensure `meta` exists and add the directive options
-          newTokens[0].meta = {
+          const [content, opts] = parseDirectiveOptions(
+            token.content.trim() ? token.content.split(/\r?\n/) : [],
+            directive
+          )
+          const directiveOpen = new state.Token("parsed_directive_open", "", 1)
+          directiveOpen.info = token.info
+          directiveOpen.hidden = true
+          directiveOpen.content = content.join("\n").trim()
+          directiveOpen.meta = {
+            arg: token.meta.arg,
+            opts
+          }
+          const newTokens = [directiveOpen]
+          newTokens.push(...directive.run(data))
+          const directiveClose = new state.Token("parsed_directive_close", "", -1)
+          directiveClose.hidden = true
+          newTokens.push(directiveClose)
+          // Ensure `meta` exists and add the directive options to parsed child
+          newTokens[1].meta = {
             directive: true,
             ...data.options,
-            ...newTokens[0].meta
+            ...newTokens[1].meta
           }
```

Comment:
As just discussed, I don't think we need this any more, since you can retrieve this info from the (directive) parent node.